daemon/containerd/progress.go

```diff
@@ -337,3 +337,21 @@ func (combined combinedProgress) UpdateProgress(ctx context.Context, ongoing *jo
 	}
 	return nil
 }
+
+// showBlobProgress determines if the progress of pulling/pushing blob should be shown.
+// Only indexes, manifests, and configs are hidden to align with the pre-containerd behavior.
+// They are small enough JSON files so it's fine to not show them.
+// We mostly care about bigger content like layers or other blobs.
+func showBlobProgress(desc ocispec.Descriptor) bool {
+	// We can't just plainly use c8dimages.IsLayerType here because it wouldn't
+	// account for other potentially big blobs like artifacts or non-standard
+	// images.
+	switch {
+	case c8dimages.IsIndexType(desc.MediaType),
+		c8dimages.IsManifestType(desc.MediaType),
+		c8dimages.IsConfigType(desc.MediaType):
+		return false
+	default:
```

Comment:
Thanks for that comment; silly question; would we still be able to take advantage of it as a fast-path? Considering;


```suggestion
	// We can't just plainly use c8dimages.IsLayerType here because it wouldn't
	// account for other potentially big blobs like artifacts or non-standard
	// images.
	switch {
	case c8dimages.IsLayerType(desc.MediaType):
		// Fast path: we always show progress for layers.	
		return true
	case c8dimages.IsIndexType(desc.MediaType),
		c8dimages.IsManifestType(desc.MediaType),
		c8dimages.IsConfigType(desc.MediaType):
		return false
	default:
```

Reply:
We can!